linkatos/activities.py

```diff
@@ -1,45 +1,55 @@
 import time
 import linkatos.parser as parser
-import linkatos.confirmation as confirmation
 import linkatos.printer as printer
-import linkatos.utils as utils
 import linkatos.firebase as fb
+import linkatos.reaction as react
 
 
-def keep_wanted_urls(expecting_confirmation, url, slack_client, BOT_ID,
-                     fb_credentials, firebase):
-    READ_WEBSOCKET_DELAY = 1  # 1 second delay between reading from firehose
+def is_empty(events):
+    return ((events is None) or (len(events) == 0))
+
+
+def is_url(url_message):
+    return url_message['type'] == 'url'
 
-    # parse the messages. Get a dictionary with @out, @channel,
-    # @out_type
-    parsed_message = parser.parse(slack_client.rtm_read(), BOT_ID)
 
-    print(parsed_message)
-    print(expecting_confirmation)
+def event_consumer(expecting_url, expecting_reaction, parsed_url_message,
+                   slack_client, fb_credentials, firebase):
+    READ_WEBSOCKET_DELAY = 1  # 1 second delay between reading from firehose
+
+    # Read slack events
+    events = slack_client.rtm_read()
 
-    if utils.is_fresh_url(expecting_confirmation, parsed_message['type']):
-        url = parsed_message['out']
+    if is_empty(events):
+        return (expecting_url, expecting_reaction, parsed_url_message)
 
-    print(url)
+    for event in events:
+        print(event)
+        print('expecting_url: ', expecting_url)
+        print('expecting_reaction: ', expecting_reaction)
+        if 'type' in event:
+            if expecting_url and event['type'] == 'message':
```

Comment:
You could add a `is_message` function in a similar way you have `is_url` or `is_empty`.